### PR TITLE
New Options : Pruner methods, Pruner Precision, ProgressivePreproc

### PR DESCRIPTION
**Pull request by lducas**
Changing Pruner Descent Methods and Pruner Precision does not seems to improve the resulting strategies, at least up to dim 75, yet those option seems relevant to future use (and are already available in fplll/fpylll pruner).

ProgressivePrerocStrategizer seems to improve performances over OneTourPreprocStrategizer by about 25% in large dimensions (75~85), and is no worse in small dimension. It is thus now set as the default strategizer.

### CODE REVIEW

File: strategizer/strategizers.py [hunk 1]
```diff
@@ -107,30 +112,67 @@ def SimplePreprocStrategizerFactory(start, stop, step_size):
...
     name = "OnePreprocStrategy-%d"%(block_size)
-    return type("OnePreprocStrategizer",
-                (OnePreprocStrategizerTemplate,),
+    return type("OneTourPreprocStrategizer",
+                (OneTourPreprocStrategizerTemplate,),
+                {"name": name, "preprocessing_block_size": block_size,
+                 "min_block_size": block_size+1})
+
+
+progressiveStep = 10
```
Comment (malb) [inline strategizer/strategizers.py]: evilCamelCase

**Comment by malb** [comment]
Can I ask you to run `flake8 --max-line-length=120 --max-complexity=16 --ignore=E22,E241` on the file (`pip install flake8` first if needed) and fix the issues reported? Looks good otherwise.

**Comment by lducas** [comment]
Will do.

2016-08-31 9:55 GMT+02:00 Martin Albrecht notifications@github.com:

File: strategizer/strategizers.py (hunk 1 above)
Comment (lducas) [reply strategizer/strategizers.py]: Sorry, what would you like me to change here ?

**Comment by lducas** [comment]
Ran flake8, and repaired. Except maybe for evilCamelCase.

**Comment by malb** [comment]
PS: You should add yourself as an author in the README.

**Comment by lducas** [comment]
In README.rst ? I've been there since the first commit I think...

**Comment by malb** [comment]
Argh, where's my head? Of course! Too many repos to keep track off. Sorry for the noise.

**Pull request merged**